### PR TITLE
feat(repo_name): Fix repository name conflicts by using owner_repo naming convention

### PR DESCRIPTION
**Pull request by XingLiu0923**
# Fix repository name conflicts by using owner_repo naming convention

## Problem
When processing repositories with the same name but different owners (e.g., `ownerA/repo` and `ownerB/repo`), the original logic in deepwiki-open could not distinguish between them. If `ownerA/repo` was processed first, subsequent requests for `ownerB/repo` would incorrectly use the documents generated from `ownerA/repo`.

## Root Cause
The original implementation used only the repository name (e.g., "repo") as the identifier for both:
- Downloaded repository directory names
- Knowledge base file names

This caused conflicts when different owners had repositories with identical names.

## Solution
Modified the `_extract_repo_name_from_url` method in `DatabaseManager` class to use the "owner_repo" naming convention:

- **GitHub URLs**: `https://github.com/owner/repo` → `owner_repo`
- **GitLab URLs**: `https://gitlab.com/owner/repo` → `owner_repo`
- **Bitbucket URLs**: `https://bitbucket.org/owner/repo` → `owner_repo`
- **Local paths**: Fallback to original behavior for backward compatibility

### Key Changes
1. **Repository Storage**: Downloaded repos now stored as `~/.adalflow/repos/{owner}_{repo}`
2. **Database Files**: Knowledge bases now saved as `~/.adalflow/databases/{owner}_{repo}.pkl`
3. **Conflict Prevention**: The owner_repo format significantly reduces the probability of naming conflicts

## Testing
- Added comprehensive unit tests in `test/test_extract_repo_name.py`
- Tests cover GitHub, GitLab, Bitbucket URLs and local paths
- All tests pass locally
- Edge cases and error conditions are properly handled

## Example
**Before**: 
- `https://github.com/octocat/Hello-World` → `Hello-World`
- `https://github.com/yankils/hello-world` → `hello-world` (potential conflict)

**After**:
- `https://github.com/octocat/Hello-World` → `octocat_Hello-World`
- `https://github.com/yankils/hello-world` → `yankils_hello-world` (no conflict)

## Discovery Context
This issue was discovered while testing with:
- `https://github.com/octocat/Hello-World`
- `https://github.com/yankils/hello-world`

The similar repository names exposed the naming conflict bug in the original implementation.

## Backward Compatibility
- Local repository paths maintain original behavior
- Existing processed repositories will need to be re-processed to benefit from the new naming scheme
- No breaking changes to the public API

### CODE REVIEW

File: api/config/embedder.json.bak [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  },
+  "retriever": {
+    "top_k": 20
+  },
+  "text_splitter": {
+    "split_by": "word",
+    "chunk_size": 350,
+    "chunk_overlap": 100
+  }
+}
```
[the remark this answers is not in the document]
Comment (XingLiu0923) [reply api/config/embedder.json.bak:19]: This is used for easily revert embedder.json after i  use other bak file as embedder.json locally, which will bring a lot of convenience I think

File: test/test_extract_repo_name.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        result = self.db_manager._extract_repo_name_from_url("https://github.com/owner/repo", "github")
+        assert result == "owner_repo"
+
+        result = self.db_manager._extract_repo_name_from_url("https://gitlab.com/group/subgroup/repo", "gitlab")
+        assert result == "subgroup_repo"
+
+        result = self.db_manager._extract_repo_name_from_url("https://bitbucket.org/owner/repo", "bitbucket")
+        assert result == "owner_repo"
+
+        print("✓ Edge case tests passed")
```
[the remark this answers is not in the document]
Comment (XingLiu0923) [reply test/test_extract_repo_name.py]: yes i'll remove

File: test/test_extract_repo_name.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        local_path = "/home/user/projects/my-repo"
+        result = self.db_manager._extract_repo_name_from_url(local_path, "local")
+        assert result == "my-repo"
+        
+        # Test local path with .git suffix
+        local_git = "/var/repos/project.git"
+        result = self.db_manager._extract_repo_name_from_url(local_git, "local")
+        assert result == "project"
+        
+        print("✓ Local path tests passed")
```
[the remark this answers is not in the document]
Comment (XingLiu0923) [reply test/test_extract_repo_name.py:103]: I'll remove redundant one
